src/methods/join.js

```diff
@@ -232,19 +251,47 @@ export class JoinImplementation {
     return new Effects(c, generator, bindings, properties, createdObjects);
   }
 
-  joinValuesOfSelectedCompletions(selector: Completion => boolean, completion: Completion): Value {
+  joinValuesOfSelectedCompletions(
+    selector: Completion => boolean,
+    completion: Completion,
+    keepInfeasiblePaths: boolean = false
```

Comment:
Why would it ever be desirable to keep infeasible paths?

Reply:
It is not desirable, but a hack to work around the fact that generator.emitConditionalThrow wants a value for its parameter. Fixing that to take the completion instead would be preferable. For this PR, however, that seems out of scope.